ledger/participant-integration-api/src/main/scala/platform/indexer/parallel/ParallelIndexerFactory.scala

```diff
@@ -75,55 +75,80 @@ object ParallelIndexerFactory {
+              connectionTimeout = FiniteDuration(
+                250,
+                "millis",
+              ), // 250 millis is the lowest possible value for this Hikari configuration (see HikariConfig JavaDoc)
+              metrics = metrics,
+            )
+        ) { dbDispatcher =>
+          initializeParallelIngestion(
+            dbDispatcher = dbDispatcher,
+            readService = readService,
+            ec = ec,
+            mat = mat,
+          ).map(
+            parallelIndexerSubscription(
+              inputMapperExecutor = inputMapperExecutor,
+              batcherExecutor = batcherExecutor,
               dbDispatcher = dbDispatcher,
-              readService = readService,
-              ec = ec,
-              mat = mat,
-            ).map(
-              parallelIndexerSubscription(
-                inputMapperExecutor = inputMapperExecutor,
-                batcherExecutor = batcherExecutor,
-                dbDispatcher = dbDispatcher,
-                materializer = mat,
-              )
-            ).andThen {
-              // the tricky bit:
-              // the future in the completion handler will be this one
-              // but the future for signaling for the HaCoordinator, that the protected execution is initialized, needs to complete precisely here
-              case Success(handle) => killSwitchPromise.success(handle.killSwitch)
-              case Failure(ex) => killSwitchPromise.failure(ex)
-            }.flatMap(_.completed)
-          }
+              materializer = mat,
+            )
+          )
+        }
+      )
+    }
 
-        killSwitchPromise.future
-          .map(Handle(completionFuture.map(_ => ()), _))
+  /** Helper function to combine a ResourceOwner and an initialization function to initialize a Handle.
+    *
+    * @param owner A ResourceOwner which needs to be used to spawn a resource needed by initHandle
+    * @param initHandle Asynchronous initialization function to create a Handle
+    * @return A Future of a Handle where Future encapsulates initialization (as completed initialization completed)
+    */
+  def initializeHandle[T](
+      owner: ResourceOwner[T]
+  )(initHandle: T => Future[Handle])(implicit rc: ResourceContext): Future[Handle] = {
+    implicit val ec: ExecutionContext = rc.executionContext
+    val killSwitchPromise = Promise[KillSwitch]()
+    val completed = owner
+      .use(resource =>
+        initHandle(resource)
+          .andThen {
+            // the tricky bit:
+            // the future in the completion handler will be this one
+            // but the future for signaling completion of initialization (the Future of the result), needs to complete precisely here
+            case Success(handle) => killSwitchPromise.success(handle.killSwitch)
+          }
+          .flatMap(_.completed)
+      )
+      .andThen {
+        // if error happens:
+        //   - at Resource initialization: result should complete with a Failure
```

Comment:
Can you be more precise here
```suggestion
        //   - at Resource initialization in owner.use: result should complete with a Failure
```

Reply:
changed together